src/NextGen/DonationForm/Registrars/resources/registrars.ts

```diff
@@ -1,54 +1,35 @@
 import GatewayRegistrar from './GatewayRegistrar';
 import FormDesignRegistrar from './FormDesignRegistrar';
 import type {FormServerExports} from '@givewp/forms/types';
-import type {useFormContext, useWatch} from 'react-hook-form';
-
-import type {
-    getElementTemplate,
-    getFieldErrorTemplate,
-    getFieldLabelTemplate,
-    getFieldTemplate,
-    getGoalTemplate,
-    getGroupTemplate,
-    getHeaderDescriptionTemplate,
-    getHeaderTemplate,
-    getHeaderTitleTemplate,
-} from '@givewp/blocks/form/app/templates';
-
-if (!window.givewp) {
-    // @ts-ignore
-    window.givewp = {};
-}
-
-window.givewp.gateways = new GatewayRegistrar();
-window.givewp.form = {
-    ...window.givewp.form,
-    designs: new FormDesignRegistrar(),
-};
+import type {useFormContext, useFormState, useWatch} from 'react-hook-form';
 
 declare global {
     interface Window {
+        giveNextGenExports: FormServerExports;
         givewp: {
             gateways: GatewayRegistrar;
-            templates: {
-                getFieldLabel: typeof getFieldLabelTemplate;
-                getFieldError: typeof getFieldErrorTemplate;
-                getField: typeof getFieldTemplate;
-                getElement: typeof getElementTemplate;
-                getGroup: typeof getGroupTemplate;
-                getHeader: typeof getHeaderTemplate;
-                getHeaderTitle: typeof getHeaderTitleTemplate;
-                getHeaderDescription: typeof getHeaderDescriptionTemplate;
-                getGoal: typeof getGoalTemplate;
-            };
             form: {
                 designs: FormDesignRegistrar;
                 hooks: {
                     useFormContext: typeof useFormContext;
                     useWatch: typeof useWatch;
+                    useFormState: typeof useFormState;
                 };
             };
         };
-        giveNextGenExports: FormServerExports;
     }
 }
+
+if (!window.givewp) {
+    // @ts-ignore
+    window.givewp = {};
+}
+
+window.givewp = {
+    ...window.givewp,
+    gateways: new GatewayRegistrar(),
+    form: {
+        ...window.givewp.form,
+        designs: new FormDesignRegistrar(),
+    },
+};
```

Comment:
This could potentially get expensive as more stuff gets shoved in here. Let's avoid rebuilding this object unless we have a clear reason. Just mutate:
```js
window.givewp.gateways = new GatewayRegistrar();
window.givewp.form.designs = new FormDesignRegistrar();
```

Reply:
resolved d005c6ac